### PR TITLE
Option to select the scale of the Y axis

### PR DESCRIPTION
**Pull request by jasoncalabrese**
screen shot 2015-10-19 at 7 23 27 pm

screen shot 2015-10-19 at 7 23 44 pm

### CODE REVIEW

**Comment by jasoncalabrese** [comment]
@kenstack  @channemann @bewest

**Comment by bewest** [comment]
Nice.

**Comment by jasoncalabrese** [comment]
@Fokko think you had a version of this before also

**Comment by MilosKozak** [comment]
Describe both options in readme

**Comment by MilosKozak** [comment]
Ticks should be different for both types. You may take it from reports

**Comment by jasoncalabrese** [comment]
good idea for the readme, suggestions for the text?   @oamyoamy?

the ticks aren't customizable yet, so setting good default for the linear scale make sense

**Comment by MilosKozak** [comment]
https://github.com/nightscout/cgm-remote-monitor/blob/master/lib/report_plugins/daytoday.js#L93

**Comment by jasoncalabrese** [comment]
:+1:

**Comment by jasoncalabrese** [comment]
@MilosKozak I refactored the ticks some so they are shared between the main chart and day-to-day.  Would be nice to mix in the targets with the linear ticks.

**Comment by MilosKozak** [comment]
:+1:

**Comment by MilosKozak** [comment]
add  custom targets to linear too. I think sorting is not necessary  so you can add to the end.
you will have to move custom targets out of ticks.js or pass them as a params. in reports are customisable ....

**Comment by oamyoamy** [comment]
Suggested readme info:   `SCALE_Y` (`log`) - The type of scaling used for the Y axis of the charts system wide. Set to 'linear' for chart scaling with equidistant tick marks. (Linear or log scaling can be altered locally via settings panel in the browser.)

**Comment by channemann** [comment]
Good stuff, thanks for the mention

**Comment by jasoncalabrese** [comment]
@MilosKozak I added support for the option used in day-to-day with the new ticks module, the 40 / 2 tick is getting cropped by the open top line with linear scaling.  I tried adjusting it, but wasn't working maybe you have an idea.  If we can't find a quick fix maybe if can be fixed later.

@oamyoamy I updated the readme for the Log/Linear options based on your suggestion and added a little more.  You could probably come up with something better.

File: lib/client/ticks.js [hunk 1]
```diff
@@ -0,0 +1,77 @@
+'use strict';
+
+function prepare (client, opts) {
```
Comment (jasoncalabrese) [inline lib/client/ticks.js]: Codacy Issue found: [This function's cyclomatic complexity is too high. (8)](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/1150973166/issues/source?bid=2452379&fileBranchId=2909529#l3)

Comment (jasoncalabrese) [inline lib/client/ticks.js]: Codacy Issue found: [This function's cyclomatic complexity is too high. (8)](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/1156649228/issues/source?bid=2452379&fileBranchId=2909529#l3)

File: lib/client/ticks.js [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+      , opts.targetBottom
+      , 120
+      , opts.targetTop
+      , opts.high
+      , 400
+    ];
+  }
+}
+
+function prepareLinear (client, opts) {
```
Comment (jasoncalabrese) [inline lib/client/ticks.js]: Codacy Issue found: ['opts' is defined but never used.](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/1159517207/issues/source?bid=2452379&fileBranchId=2909529#l50)

Comment (jasoncalabrese) [inline lib/client/ticks.js]: Codacy Issue found: ['opts' is defined but never used.](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/1159572931/issues/source?bid=2452379&fileBranchId=2909529#l50)

**Comment by jasoncalabrese** [comment]
A new option for dynamic ranges with the log scale
screen shot 2015-10-21 at 8 15 35 pm

**Comment by MilosKozak** [comment]
MAYBE with dynamic scale would be nice to adjust basal scale too. to not overlap but use 100% of space
https://github.com/nightscout/cgm-remote-monitor/blob/dev/lib/client/chart.js#L209
https://github.com/nightscout/cgm-remote-monitor/pull/1222/files#diff-6cf944f042d6214695a022aad3d04f39R46

**Pull request merged**